lib/matplotlib/colors.py

```diff
@@ -955,6 +955,174 @@ def scaled(self):
         return self.vmin is not None and self.vmax is not None
 
 
+class PiecewiseLinearNorm(Normalize):
+    """
+    Normalizes data into the ``[0.0, 1.0]`` interval over linear segments.
+    """
+    def __init__(self, data_points, norm_points=None):
+        """Normalize data linearly between the defined stop points.
+
+        Parameters
+        ----------
+        data_points : tuple of floats
+            Floats spanning the data to be mapped.
+
+        norm_points : tuple of floats or None (default)
+            Floats spanning [0, 1] that the data points will map to.  If
+            *None*, the data points will be mapped equally into [0, 1].
+
+        Examples
+        --------
+        Note this example is equivalent to the `.DivergingNorm` example.
+        >>> import matplotlib.colors as mcolors
+        >>> offset = mcolors.PiecewiseLinearNorm([-2, 0, 4])
+        >>> data = [-2., -1., 0., 1., 2., 3., 4.]
+        >>> offset(data)
+        array([0., 0.25, 0.5, 0.625, 0.75, 0.875, 1.0])
+        """
+        if not data_points:
+            raise ValueError("data_points must increase monotonically")
+
+        self._data_points = np.array(data_points)
+        if norm_points is not None:
+            self._norm_points = np.array(norm_points)
+        else:
+            self._norm_points = None
+
+        if not np.all(np.diff(self._data_points) > 0):
+            raise ValueError("data_points must increase monotonically")
+
+        self.vmin = data_points[0]
```

Comment:
make these properties? or does it make sense for vmin/vmax to be updated independently of data_points?